pkg/sensors/program/loader.go

```diff
@@ -22,19 +22,29 @@ var (
 
 // AttachFunc is the type for the various attachment functions. The function is
 // given the program and it's up to it to close it.
-type AttachFunc func(*ebpf.Program, *ebpf.ProgramSpec) (unloader.Unloader, error)
+type AttachFunc func(*ebpf.Collection, *ebpf.CollectionSpec, *ebpf.Program, *ebpf.ProgramSpec) (unloader.Unloader, error)
+
+type OpenFunc func(*ebpf.CollectionSpec) error
 
 type customInstall struct {
 	mapName   string
 	secPrefix string
 }
 
+type loadOpts struct {
+	attach AttachFunc
+	open   OpenFunc
```

Comment:
Can we add a comment here to explain what this does?
Below is my best guess:

```suggestion
         // open allows modifying the ebpf.CollectionSpec before the collection is loaded into the kernel.
         // It can be used, for example, to disable loading of certain programs.
	open   OpenFunc
```

Reply:
right, I'll add comment to loadProgram function and descirbe that in more details